gslab_scons/size_warning.py

```diff
@@ -0,0 +1,111 @@
+import os
+import re
+import sys
+import subprocess
+from _exception_classes import ReleaseError
+
+
+def issue_size_warnings(look_in = ['source', 'raw', 'release'],
+                        file_MB_limit = 2, 
+                        total_MB_limit = 500):
+    '''Issue warnings if versioned files are large'''
+    bytes_in_MB = 1000000
+    # Compile a list of files that are not versioned.
+    ignored = list_ignored_files(look_in)
+    versioned = create_size_dictionary(look_in)
+    versioned = {k: versioned[k] for k in versioned.keys() if k not in ignored}
+
+    for file_name in versioned.keys():
+        size  = versioned[file_name]
+        limit = file_MB_limit * bytes_in_MB
+
+        if size > limit and file_name:
+            print "Warning: the versioned file %s is larger than %d MB.\n" \
+                  % (file_name, file_MB_limit)
+            print "Versioning files of this size is discouraged.\n" 
+
+            response = raw_input("Would you like to continue anyway? (y|n)\n")
+            if response in ['N', 'n']: 
+                sys.exit()
+
+    total_size  = sum(versioned.values())
+    total_limit = total_MB_limit * bytes_in_MB
+
+    if total_size > total_limit:
+        print "Warning: the versioned files in /release/ are together " + \
+            "larger than " + str(total_MB_limit) + " MB.\n" + \
+            "Versioning this much content is discouraged.\n"
+        response = raw_input("Would you like to continue anyway? (y|n)\n")
+        if response in ['N', 'n']: 
+            sys.exit()
+
+
+def list_ignored_files(look_in):
+    '''List files ignored by git 
+
+    This function returns a list of the files in the directory
+    given by the dir_path argument that are ignored by git. 
+    '''
+    # Produce a listing of files and directories ignored by git
+    message = subprocess.check_output('git status --ignored', shell = True)
+    message = message.split('\n')
+    message = map(lambda s: s.strip(), message)
+
+    # Allow for the root to be in look_in
+    look_in = [os.path.normpath(d) for d in look_in]
+    if '.' in look_in:
+        # So regex searches will find the root at the beginning of all paths
+        look_in[look_in.index('.')] = ''
+
+    try:
+        ignored_line = message.index('Ignored files:')
+    except ValueError:
+        # If the git status message lists no ignored files, return empty list
+        return []
+
+    # Loop through the listing of ignored paths, classifying them
+    # as files or directories
+    ignore_dirs  = []
+    ignore_files = []
+    for line in message[ignored_line:len(message)]:
+        as_path = os.path.normpath(line)
+
+        # Only check for ignored files in the specified "look_in" directories
```

Comment:
I just wasn't sure that we'd want to wait  to list obviously unversioned files (below is for `congress_text`).
```py
>>> import gslab_scons
>>> import time
>>> t = time.time(); build_files = gslab_scons.size_warning.list_ignored_files('build'); elapsed = time.time() - t
>>> print elapsed
42.23977017402649
```
Forty-two seconds isn't too long, so I'm happy to reconsider if you'd like.